primefaces/src/main/java/org/primefaces/util/FailableRunnable.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009-2024 PrimeTek Informatics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.util;
+
+/**
+ * A functional interface like {@link Runnable} that declares a kind of {@link Throwable}.
+ *
+ * @param <E> The kind of {@link Throwable}.
+ */
+@FunctionalInterface
+public interface FailableRunnable<E extends Throwable> {
```

Comment:
We could acually shaded few of these, I'm thinking about IOBiConsumer

Reply:
I don't have thoughts other than with like our COmmons IO issue its sometimes easier not to shade?  But I could be convinced either way 😄

Reply:
Yeah... The difference I see here is that, as they are functional interfaces, the chances for them to change is very low?

Reply:
Another option would be to do like in omnifaces: https://github.com/omnifaces/omnifaces/blob/4.x/src/main/java/org/omnifaces/util/Callback.java

So we don't end up with too many java files?

Reply:
Oohhh I like that let me update and see if you like it.

Reply:
OK submitted that to a new class called `Callbacks`.

Reply:
Looks nice, at this point, you can:
- rename IOBiConsumer into FailableBiConsumer
- move SerializableXXX classes into Callbacks
- rename ConsumerThree into TriConsumer and move it also

Reply:
Done!

Reply:
Yup, looks good! WDYT?

Reply:
I like it. I think you are right I need a private CTOR